events/handler.js

```diff
@@ -1,14 +1,15 @@
 'use strict';
 
-const AWS = require('aws-sdk')
-const uuid = require('uuid')
+import Processor from './processor';
 
-module.exports.errors = (event, context, callback) => {
-  event.Records.forEach((record) => {
-    var buf = new Buffer(record.kinesis.data, 'base64').toString('ascii');
-    var data = JSON.parse(buf); 
-    console.log("Log data received = " + JSON.stringify(data))
-  });
-
-  callback(null, "Success");
+export function processEvent(event, context, callback) {
+  const processor = new Processor(event);
+  try {
+    processor.process((success, errors) => {
+      const msg = "Success = " + success.length + ", Errors = " + errors.length;
+      callback(null, msg);
+    });
+  } catch (e) {
```

Comment:
I'm not sure this will catch the errors you think it will from the async function but I might be misinterpreting the code: https://nodejs.org/api/errors.html#errors_node_js_style_callbacks